src/checkers/inference/solver/backend/SolverType.java

```diff
@@ -0,0 +1,79 @@
+package checkers.inference.solver.backend;
+
+import java.lang.reflect.Constructor;
+import java.util.Collection;
+import java.util.Map;
+
+import javax.annotation.processing.ProcessingEnvironment;
+
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.ErrorReporter;
+
+import checkers.inference.model.Constraint;
+import checkers.inference.model.Serializer;
+import checkers.inference.model.Slot;
+import checkers.inference.solver.backend.logiql.LogiQLFormatTranslator;
+import checkers.inference.solver.backend.logiql.LogiQLSolver;
+import checkers.inference.solver.backend.maxsat.LingelingSolver;
+import checkers.inference.solver.backend.maxsat.MaxSatFormatTranslator;
+import checkers.inference.solver.backend.maxsat.MaxSatSolver;
+import checkers.inference.solver.frontend.Lattice;
+
+public enum SolverType {
+
+    MAXSAT("MaxSAT", MaxSatSolver.class, MaxSatFormatTranslator.class),
+    LINGELING("Lingeling", LingelingSolver.class, MaxSatFormatTranslator.class),
+    LOGIQL("LogiQL", LogiQLSolver.class, LogiQLFormatTranslator.class);
+
+    public final String simpleName;
+    public final Class<? extends SolverAdapter<?>> solverAdapterClass;
+    public final Class<? extends FormatTranslator<?, ?, ?>> translatorClass;
+
+    private SolverType(String simpleName, Class<? extends SolverAdapter<?>> solverAdapterClass,
+            Class<? extends FormatTranslator<?, ?, ?>> translatorClass) {
+        this.simpleName = simpleName;
+        this.solverAdapterClass = solverAdapterClass;
+        this.translatorClass = translatorClass;
+    }
+
+    public FormatTranslator<?, ?, ?> createDefaultFormatTranslator(Lattice lattice) {
+        Constructor<?> cons;
+        try {
+            cons = translatorClass.getConstructor(Lattice.class);
+            return (FormatTranslator<?, ?, ?>) cons.newInstance(lattice);
+        } catch (Exception e) {
+            ErrorReporter.errorAbort(
+                    "Exception happends when creating default serializer for " + simpleName + " backend.", e);
```

Comment:
"happends" ?

Reply:
Ok, I google it, and found it should be "happens". Regarding English errors, would you kindly pointed out the fix as well? Thanks!